File: 299. Bulls and Cows/Python/Bulls and Cows.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        counts = [0] * 10
        bulls = 0
        cows = 0
        for a, b in zip(secret, guess):
            if a == b:
                bulls += 1
            else:
                if counts[ord(a) - ord('0')] < 0:
                    cows += 1
                counts[ord(a) - ord('0')] += 1
                if counts[ord(b) - ord('0')] > 0:
                    cows += 1
                counts[ord(b) - ord('0')] -= 1
        return '%dA%dB' % (bulls, cows)

File: 299. Bulls and Cows/Java/Bulls and Cows.java
public class Solution {
    public String getHint(String secret, String guess) {
        int[] counts = new int[10];
        int bulls = 0;
        int cows = 0;
        for (int i = 0; i < secret.length(); i++) {
            if (secret.charAt(i) == guess.charAt(i)) {
                bulls++;
            } else {
                if (counts[secret.charAt(i) - '0'] < 0) {
                    cows++;
                }
                counts[secret.charAt(i) - '0']++;
                if (counts[guess.charAt(i) - '0'] > 0) {
                    cows++;
                }
                counts[guess.charAt(i) - '0']--;
            }
        }
        return String.format("%dA%dB", bulls, cows);
    }
}